src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -257,14 +269,37 @@ private void dispatchPending() {
     }
 
     private void fireNextRequest() {
-        final Request request = _pendingRequests.poll();
+        final RequestEntry requestEntry = _pendingRequests.poll();
+        final Metrics metrics = _metricsFactory.create();
+        metrics.setTimer(_inQueueLatencyName, System.currentTimeMillis() - requestEntry.getEnterTime(), TimeUnit.MILLISECONDS);
+
+        final Request request = requestEntry.getRequest();
         _inflightRequestsCount++;
 
         final CompletableFuture<Response> promise = new CompletableFuture<>();
+        metrics.startTimer(_requestLatencyName);
         _client.executeRequest(request, new ResponseAsyncCompletionHandler(promise));
         final CompletionStage<Object> responsePromise = promise
-                .<Object>thenApply(response -> new PostComplete(request, response))
-                .exceptionally(PostFailure::new);
+                .handle((result, err) -> {
+                        metrics.stopTimer(_requestLatencyName);
+                        final int responseStatusCode = result.getStatusCode();
+                        final boolean requestSuccess = err == null && ACCEPTED_STATUS_CODES.contains(responseStatusCode);
+                        metrics.incrementCounter(_requestSuccessName, requestSuccess ? 1 : 0);
+
+                        if (err != null) {
+                            metrics.close();
+                            return new PostFailure(err);
```

Comment:
Can you include the the request in PostFailure too?

(part of the reason to drive the refactoring here is to set us up to retries)